crates/mun_language_server/src/lib.rs

```diff
@@ -79,24 +98,26 @@ pub async fn run_server_async() -> Result<()> {
             config.watcher = FilesWatcher::Client;
         }
 
-        // Get the current working directory as fallback
-        let cwd = std::env::current_dir()?;
-        // Convert the root uri to a PathBuf
-        let root = initialize_params
-            .root_uri
-            .and_then(|it| it.to_file_path().ok())
-            .unwrap_or(cwd);
         // Convert the workspace_roots, if these are empy use the root_uri or the cwd
-        config.workspace_roots = initialize_params
+        let workspace_roots = initialize_params
             .workspace_folders
             .map(|workspaces| {
                 workspaces
                     .into_iter()
                     .filter_map(|it| it.uri.to_file_path().ok())
+                    .filter_map(|path| AbsPathBuf::try_from(path).ok())
```

Comment:
Can't this be a single `a(path).ok() && b(path).ok()`?

Reply:
No 'ok' returns an Option. We use the result of the first operation in the next